manual/src/refman/extensions/gadts.etex

```diff
@@ -10,315 +15,19 @@ type-param:
         | [variance] '_'
 \end{syntax}
 
-Generalized algebraic datatypes, or GADTs, extend usual sum types in
-two ways: constraints on type parameters may change depending on the
-value constructor, and some type variables may be existentially
-quantified.
-Adding constraints is done by giving an explicit return type
-(the rightmost @typexpr@ in the above syntax), where type parameters
-are instantiated.
-This return type must use the same type constructor as the type being
-defined, and have the same number of parameters.
-Variables are made existential when they appear inside a constructor's
-argument, but not in its return type.
-
-Since the use of a return type often eliminates the need to name type
-parameters in the left-hand side of a type definition, one can replace
-them with anonymous types @"_"@ in that case.
-
-The constraints associated to each constructor can be recovered
-through pattern-matching.
-Namely, if the type of the scrutinee of a pattern-matching contains
-a locally abstract type, this type can be refined according to the
-constructor used.
-These extra constraints are only valid inside the corresponding branch
-of the pattern-matching.
-If a constructor has some existential variables, fresh locally
-abstract types are generated, and they must not escape the
-scope of this branch.
-
-\lparagraph{p:gadts-recfun}{Recursive functions}
-
-Here is a concrete example:
-\begin{caml_example*}{verbatim}
-type _ term =
-  | Int : int -> int term
-  | Add : (int -> int -> int) term
-  | App : ('b -> 'a) term * 'b term -> 'a term
+Refuation cases. (Introduced in OCaml 4.03)
```

Comment:
Refuation ⇒ Refutation

Reply:
Fixed. By the way, your Stackoverflow answer here could be repurposed to improve the refutation explanation in the tutorial, if you have any time to do it:

https://stackoverflow.com/questions/52795191/when-are-refutation-cases-necessary-in-ocaml

Reply:
It is a good idea, I will have a look.